src/transformers/modeling_prophetnet.py

```diff
@@ -0,0 +1,1461 @@
+            self,
+            input_ids,
+            attention_mask=None,
+            decoder_input_ids=None,
+            encoder_outputs: Optional[Tuple] = None,
+            decoder_cached_states=None,
+            use_cache=False,
+            output_attentions=None,
+    ):
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+
+
+        assert decoder_input_ids is not None
+
+        if encoder_outputs is None:
+            encoder_outputs = self.encoder(
+                input_ids=input_ids, attention_mask=attention_mask, output_attentions=output_attentions,
+            )
+        assert isinstance(encoder_outputs, tuple)
+        # decoder outputs consists of (dec_features, layer_state, dec_hidden, dec_attn)
+        decoder_outputs = self.decoder(
+            decoder_input_ids,
+            encoder_outputs[0],
+            attention_mask,
+            decoder_cached_states=decoder_cached_states,
+            output_attentions=output_attentions,
+            use_cache=use_cache,
+        )
+
+        # Attention and hidden_states will be [] or None if they aren't needed
+        decoder_outputs: Tuple = _filter_out_falsey_values(decoder_outputs)
+        assert isinstance(decoder_outputs[0], tuple) # last layer hidden state of main stream and predicting streams
+        encoder_outputs: Tuple = _filter_out_falsey_values(encoder_outputs)
+        return decoder_outputs + encoder_outputs
+
+
+def _reorder_buffer(attn_cache, new_order):
+    for k, input_buffer_k in attn_cache.items():
+        if input_buffer_k is not None:
+            attn_cache[k] = input_buffer_k.index_select(0, new_order)
+    return attn_cache
+
+def _make_linear_from_emb(emb):
+    vocab_size, emb_size = emb.weight.shape
+    lin_layer = nn.Linear(vocab_size, emb_size, bias=False)
+    lin_layer.weight.data = emb.weight.data
+    return lin_layer
+
+@add_start_docstrings(
+    "The ProphetNet Model with a language modeling head. Can be used for summarization.", PROPHETNET_START_DOCSTRING
+)
+class ProphetNetForConditionalGeneration(ProphetNetPreTrainedModel):
+    base_model_prefix = "model"
+
+    def __init__(self, config: ProphetNetConfig):
+        super().__init__(config)
+        base_model = ProphetNetModel(config)
+        self.model = base_model
+        #self.padding_idx = config.pad_token_id
+        self.padding_idx = -100
+        self.disable_ngram_loss = config.disable_ngram_loss
+
+    def output_layer(self, features, **kwargs):
+        return F.linear(features, self.model.embed_tokens.weight)
+
+    @add_start_docstrings_to_callable(PROPHETNET_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        input_ids,
+        attention_mask=None,
+        encoder_outputs=None,
+        decoder_input_ids=None,
+        decoder_cached_states=None,
+        labels=None,
+        use_cache=False,
+        output_attentions=None,
+        **unused,
+    ):
+        if "lm_labels" in unused:
```

Comment:
You can delete the whole `if "lm_labels" ... ` -> since `lm_labels` was never there in the first place we should not allow the user to use it